### DISCUSSION TITLE
how to generate an ssh key with passphrase

### DISCUSSION THREAD

**Discussion by sunhasgothishaton** [borgbase/vorta#1276] [General]
When I generate an ssh key using Vorta, how do I add a passphrase?

Is that possible? Or do I need to generate the key with passphrase using the command line first?

**Comment by m3nu** [comment]
No, we only cover the most basic use case for anyone not comfortable with the terminal. For more complex cases, you would need to get going on the terminal yourself.

**Reply by sunhasgothishaton** [reply]
Great, thanks for your reply. I'm finding vorta great as its a nice way to start with borg, which can a overload to begin with.

If I create my own key using ssh-keygen, should it appear in the list of keys in vorta? As the key I created using the command line doesn't appear in vorta.

**Comment by m3nu** [comment]
Vorta searches for valid private keys in `~/.ssh`. But only on startup, I think.

**Reply by sunhasgothishaton** [reply]
When I closed the vorta window I didn't realise it kept  running as a background process. I rebooted my computer and the keys (without passphrases) were listed.

The list of keys also refreshed when a new key is generated with vorta.

**Reply by sunhasgothishaton** [reply]
Is there a way to exit and restart vorta without rebooting?
I'm using the flatpak version.

**Reply by sunhasgothishaton** [reply]
Fantastic, got it working with a ssh key with a passphrase. It doesn't appear in the list of keys in vorta, vorta uses the correct key though.

Thanks for your help.

**Reply by m3nu** [reply]
Yes, if you leave it just at "Default" it will work like any other SSH application.

**Reply by real-yfprojects** [reply]
> Is there a way to exit and restart vorta without rebooting? I'm using the flatpak version.

You can close it via the tray icon.

**Comment by sunhasgothishaton** [comment]
Thanks @real-yfprojects, I didn't realise the icon in the system tray is supposed to have a menu. I use XFCE 4.14 and initially the menu doesn't appear. I changed to a different system notifier and now the menu works.

Thanks for your help.

**Reply by m3nu** [reply]
Yeah, there is an open issue for XFCE. It doesn't open the menu like other desktops.

**Reply by real-yfprojects** [reply]
>I changed to a different system notifier and now the menu works.

Which system notifier did you switch to?

**Reply by sunhasgothishaton** [reply]
Thanks @m3nu , I didn't notice the open ticket for XFCE.

**Reply by sunhasgothishaton** [reply]
I have installed xfce4-sntray-plugin. The menu doesn't appear on the first click of the vorta icon, after the first click the menu works correctly.

**Reply by real-yfprojects** [reply]
> I have installed xfce4-sntray-plugin. The menu doesn't appear on the first click of the vorta icon, after the first click the menu works correctly.

Though the menu isn't updated which it would happen on other DEs e.g. when a new repo is added.
